File: RepoTransBench/target_projects/Python/C++/vijos_jd4/public_tests/test_public_compare.cpp
#include <gtest/gtest.h>
#include "jd4/compare.h"

TEST(PublicCompareTest, StripTrailingSpacesNewlines)
{
    std::string s = "hello world    \n  \n\t";
    EXPECT_EQ(strip_trailing_spaces_newlines(s), "hello world");
}

TEST(PublicCompareTest, ComparePublicEqualNorm)
{
    std::string a = "foo bar   \n";
    std::string b = "foo bar";
    bool eq = compare_stream(a, b);
    EXPECT_TRUE(eq);
}

TEST(PublicCompareTest, ComparePublicNotEqual)
{
    std::string a = "value1";
    std::string b = "value2";
    bool eq = compare_stream(a, b);
    EXPECT_FALSE(eq);
}

File: RepoTransBench/target_projects/Python/Rust/vijos_jd4/public_tests/test_public_compare.rs
use vijos_jd4::compare::{compare, strip_trailing_spaces_newlines};

#[test]
fn test_strip_trailing_spaces_newlines_public() {
    let s = "hello world    \n  \n\t";
    assert_eq!(strip_trailing_spaces_newlines(s), "hello world");
}

#[test]
fn test_compare_public_equal_norm() {
    let a = "foo bar   \n";
    let b = "foo bar";
    let eq = compare(a, b, true);
    assert!(eq);
}

#[test]
fn test_compare_public_not_equal() {
    let a = "value1";
    let b = "value2";
    let eq = compare(a, b, false);
    assert!(!eq);
}